tests/testthat/test-verb-joins.R

```diff
@@ -72,6 +72,42 @@ test_that("join works with in_schema", {
   )
 })
 
+test_that("alias truncates long table names at database limit", {
+  # Postgres has max table name length of 63; ensure it's not exceeded when generating table alias
+  withr::local_db_connection(con <- dbConnect(RSQLite::SQLite(), ":memory:"))
+
+  nm1 <- paste0("a", paste0(0:62 %% 10, collapse = ""))
+  DBI::dbWriteTable(con, nm1, tibble(x = 1:3, y = "a"))
+  mf1 <- tbl(con, nm1)
+
+  nm2 <- paste0("b", paste0(0:62 %% 10, collapse = ""))
+  DBI::dbWriteTable(con, nm2, tibble(x = 2:3, y = "b"))
+  mf2 <- tbl(con, nm2)
+
+  # 2 tables
+  self_join2 <- left_join(mf1, mf1, by = c("x", "y"))
+
+  expect_equal(
+    self_join2 %>% collect(),
+    tibble(x = 1:3, y = "a")
+  )
+  expect_snapshot(
```

Comment:
Given that you want to make specific assertions about the length of the generate table names, is it possible to avoid the snapshot tests in favour of something more specific?

Reply:
Good thinking! Added some to look at the actual values from `generate_join_table_names()`; this catches the error observed above.